autode/neb/idpp.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+from scipy.spatial import distance_matrix
+
+
+class IDPP:
+    """
+    Image dependent pair potential (IDPP) objective function from
+    https://arxiv.org/pdf/1406.1512.pdf
+
+    .. math::
+
+        S = Σ_i  Σ_{j>i} w(r_{ij}) (r_{ij}^k - r_{ij})^2
+
+    where :math:`r_{ij}` is the distance between atoms i and j and
+    :math:`r_{ij}^k = r_{ij}^0 + k(r_{ij}^N - r_{ij}^0)/N` for :math:`N` images.
+    Note: here the superscripts denote the image number, not a power. The
+    weight function is :math:`w(r_ij) = r_{ij}^{-4}`, as suggested in the paper
+    """
+
+    def __init__(self, images: 'autode.neb.original.Images'):
+        """Initialise a IDPP potential"""
+
+        if len(images) < 2:
+            raise ValueError('Must have at least 2 images for IDPP')
+
+        # Distance matrices containing all r_{ij}^k
+        self._dists = {image.name: None for k, image in enumerate(images)}
```

Comment:
unecessary `enumerate` here?

Reply:
Unnecessary but explanatory (?) when read with the maths above